internal/bminventory/inventory.go

```diff
@@ -2289,6 +2291,160 @@ func (b *bareMetalInventory) DownloadHostLogs(ctx context.Context, params instal
 	return filemiddleware.NewResponder(installer.NewDownloadHostLogsOK().WithPayload(respBody), downloadFileName, contentLength)
 }
 
+func (b *bareMetalInventory) CreateClusterManifest(ctx context.Context, params installer.CreateClusterManifestParams) middleware.Responder {
+	log := logutil.FromContext(ctx, b.log)
+	if params.CreateManifestParams.Folder == nil {
+		defaultFolder := models.CreateManifestParamsFolderManifests
+		params.CreateManifestParams.Folder = &defaultFolder
+	}
+
+	fileName := filepath.Join(*params.CreateManifestParams.Folder, *params.CreateManifestParams.FileName)
+	log.Infof("Creating manifest %s in cluster %s", fileName, params.ClusterID)
+
+	var cluster common.Cluster
+	if err := b.db.First(&cluster, identity.AddUserFilter(ctx, "id = ?"), params.ClusterID).Error; err != nil {
+		log.WithError(err).Errorf("failed to get cluster %s", params.ClusterID)
+		return common.NewApiError(http.StatusNotFound, err)
+	}
+
+	manifestContent, err := base64.StdEncoding.DecodeString(*params.CreateManifestParams.Content)
+	if err != nil {
+		log.Errorf("Cluster manifest %s for cluster %s failed to base64 decoded: [%s]", fileName, cluster.ID, *params.CreateManifestParams.Content)
+		return installer.NewCreateClusterManifestBadRequest().
+			WithPayload(common.GenerateError(http.StatusBadRequest, errors.New("failed to base64-decode cluster manifest content")))
+	}
+
+	objectName := filepath.Join(string(*cluster.ID), fileName)
+	exists, err := b.objectHandler.DoesObjectExist(ctx, objectName)
+	if err != nil {
+		log.WithError(err).Errorf("Failed to create cluster manifest")
+		return installer.NewCreateClusterManifestInternalServerError().
+			WithPayload(common.GenerateError(http.StatusInternalServerError, err))
+	}
+
+	if exists {
+		msg := fmt.Sprintf("Cluster manifest %s already exists in cluster %s", fileName, cluster.ID)
+		log.Warn(msg)
+		return installer.NewCreateClusterManifestConflict().WithPayload(common.GenerateError(http.StatusConflict, errors.New(msg)))
+	}
```

Comment:
Why not allow over-writing?  No extra logic required, should just work.

Reply:
+1